cpp/src/io/parquet/page_data.cu

```diff
@@ -51,6 +51,12 @@ constexpr int non_zero_buffer_size = block_size * 2;
 
 constexpr int rolling_index(int index) { return index & (non_zero_buffer_size - 1); }
 
+// the PageNestingDecodeInfo struct is 48 bytes (as of 1/18/23) so
+// this cache costs 480 bytes per block. Columns nested more than 2 or 3 deep are exceedingly
+// rare.  If we encounter one that exceeds the below limit, the code falls back on the uncached
+// PageNestingDecodeInfo stored in global memory.
+constexpr int max_cacheable_nesting_decode_info = 10;
```

Comment:
Somewhat arbitrary.  I feel like 480 bytes is a reasonable amount to reserve out of shared (1%) and 10 levels of nesting is far beyond typical.  I could go higher but since this kernel uses a small number of warps (4 max per block), by leaving more shared memory available it helps allow other stuff to be scheduled at the same time.

Reply:
maybe then it should be `512/sizeof(PageNestingDecodeInfo)` (assuming 512 is the actual limit you don't want to go over)